src/main/java/io/numaproj/numaflow/sideinput/Message.java

```diff
@@ -4,35 +4,31 @@
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 
-/**
- * Message is used to wrap the data returned by Side Input Retriever.
- */
-
+/** Message is used to wrap the data returned by Side Input Retriever. */
 @Getter
 @AllArgsConstructor(access = AccessLevel.PRIVATE)
-public class Message {
-    private final byte[] value;
-    private final boolean noBroadcast;
+public final class Message {
+  private final byte[] value;
+  private final boolean noBroadcast;
 
-    /**
-     * createBroadcastMessage creates a new Message with the given value
-     * This is used to broadcast the message to other side input vertices.
-     *
-     * @param value message value
-     *
-     * @return returns the Message with noBroadcast flag set to false
-     */
-    public static Message createBroadcastMessage(byte[] value) {
-        return new Message(value, false);
-    }
+  /**
+   * createBroadcastMessage creates a new Message with the given value This is used to broadcast the
+   * message to other side input vertices.
+   *
+   * @param value message value
+   * @return returns the Message with noBroadcast flag set to false
+   */
+  public static Message createBroadcastMessage(byte[] value) {
+    return new Message(value, false);
```

Comment:
Does `@AllArgsConstructor` takes care of copying here?

Reply:
No it doesn't. Good catch, I will search all `@AllArgsConstructor` and apply defensive copying as needed.